src/Microsoft.NetCore.Analyzers/CSharp/Performance/CSharpMutableStructsShouldNotBeUsedForReadonlyFields.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Performance;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Performance
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpMutableStructsShouldNotBeUsedForReadonlyFieldsAnalyzer : MutableStructsShouldNotBeUsedForReadonlyFieldsAnalyzer
+    {
+        protected override void RegisterDiagnosticAction(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(NodeAction, SyntaxKind.FieldDeclaration);
+        }
+
+        private static void NodeAction(SyntaxNodeAnalysisContext context)
+        {
+            var fieldDeclaration = (FieldDeclarationSyntax)context.Node;
+
+            var fieldTypeInfo = context.SemanticModel.GetTypeInfo(fieldDeclaration.Declaration.Type);
+
+            if (!(fieldTypeInfo.Type is INamedTypeSymbol fieldType))
+            {
+                return;
+            }
+
+            var typesToCheck = MutableValueTypesOfInterest.Select(typeName => context.Compilation.GetTypeByMetadataName(typeName)).Where(type => type != null).ToList();
```

Comment:
`GetTypeByMetadataName` should only be invoked during CompilationStartAction callback. You should check for null values of the required well known type(s) and bail out if the analyzer is not applicable for the current compilation, and register nested symbol action if it is applicable. For example: https://github.com/dotnet/roslyn-analyzers/blob/3e4ef017b0a83b1330527b35aa2781bbca927e0d/src/Microsoft.NetCore.Analyzers/Core/Tasks/DoNotCreateTasksWithoutPassingATaskScheduler.cs#L42-L54